### ISSUE TITLE
REL-1 — Migration guide v0.x→v1, DROP list, and the AI-assistant migration skill

### ISSUE THREAD

**Issue by aacostadiaz** [ACEsuit/mace#1603]
**Depends on:** FM-2 (#1587), CLI-1 (#1579) · **Blocks:** REL-2 (#1604), REL-3 (#1605), RET-6 (#1602)

**Prerequisite:** the Phase 0 characterization suite is committed — goldens, fixtures and the single-source tolerance table under `tests/golden/`, characterization tests under `tests/unit/`. References of the form `P0-N` name that work.

**Context:** v1 is a deliberate architectural break, not a drop-in upgrade, and users need one authoritative document that says so plainly and tells them exactly what to do. Two facts drive every user question and must be stated loudly and up front:

1. **User checkpoints from v0.3.x are NOT runtime-loadable in v1.** Only *published foundation models* get a converter, and even that converter is **one-shot** (offline), not a runtime loader. The compatibility guarantee that "foundation models remain loadable" means **one-shot convertible**, not runtime-loadable — this ticket is where that redefinition becomes user-visible.
2. **The public Python import surface is gone by design.** `mace.modules.MACE`, `mace.data.AtomicData`, `mace.cli.run_train.run(args)`, the string→class registries (`interaction_classes`, `readout_classes`, `scaling_classes`, `gate_dict` in `mace/modules/__init__.py`), `mace.calculators.MACECalculator`, `mace.calculators.MagneticMACECalculator` etc. do not exist in v1; v1 exposes a new public API under `mace_core`/`mace_torch` (built by the CORE/ARCH tickets). Downstream code that imports the old surface must be ported.

This ticket produces the user-facing migration story: the guide and the AI-assistant skill. It ships **no release** — the v0.3.x line ships from `main` in the ordinary way until v1, so there is no legacy artifact left to publish here and no branch to create. One consequence to state plainly in the guide, because it cannot be fixed by documentation: **no deprecation warning will ever fire in a user's terminal** naming a v1 equivalent — the last 0.3.x binary predates v1's CLI. The guide is the only signpost, so it has to be findable from the README, the docs landing page and the PyPI description. It consumes, not invents: the DROP/MERGE/KEEP dispositions come from the P0-0 functionality inventory; the converter is the FM-1 (#1586) end-user tool; the command mapping is the CLI-1 (#1579) surface. Legacy toolchain being retired is cited by path: the **12** `console_scripts` at `setup.cfg:43-55` and the **13** default property keys in `mace/tools/default_keys.py`.

**Interface & constraints:**

**A. The one-shot conversion story.** The migration guide's model section documents exactly one supported path for pretrained weights, using the FM-1 (#1586) end-user converter: a pinned `mace-torch==0.3.17` venv reads the old pickle → emits neutral **safetensors + JSON** → imports into v1 with full metadata, verified against the FM-2 (#1587) goldens. It applies **only** to the published artifacts on the settled list — the 25 the legacy loaders reach: **16 MACE-MP aliases** (`small`/`medium`/`large`, the `-0b`/`-0b2`/`-0b3` tiers, `medium-mpa-0`, `small-omat-0`/`medium-omat-0`, `mace-matpes-pbe-0`, `mace-matpes-r2scan-0`, and `mh-0`/`mh-1`), **3 MACE-OFF** (`small`/`medium`/`large`), **MACE-MDP**, **MACE-OMOL** (`extra_large`), and — from ELEC-2 (#1592) — the **3 MACE-Polar** sizes (`polar-1-s`/`-m`/`-l`). **MACE-ANI-CC is dropped** — superseded by MACE-OFF, so the guide points its users there; note it is the one artifact *bundled inside the installed package today* (`mace/calculators/foundations_models/ani500k_large_CC.model`), so its users lose it by uninstalling rather than by a failed download, and the guide must say so. **Arbitrary user checkpoints are out of scope for conversion**; the guide says so explicitly and points those users at retraining under `mace train` or staying on their pinned v0.3.x release.

**B. CLI migration map (every v0.3.x command → v1).** The guide must map *every* command; the settled table covers all 12 `console_scripts` entries at `setup.cfg:43-55` (v1 command surface owned by CLI-1 (#1579), reached via the `--engine v1` launcher):

| v0.3.x command | v1 equivalent | Disposition / note |
|---|---|---|
| `mace_run_train` | `mace train` | KEEP — new config system |
| `mace_eval_configs` | `mace eval` | KEEP — thin adapter over the ASE calculator |
| `mace_prepare_data` | `mace data prepare` | KEEP |
| `mace_create_lammps_model` | `mace export lammps` | KEEP — MLIAP-first bundle, no TorchScript |
| `mace_select_head` | `mace model select-head` | KEEP |
| `mace_plot_train` | `mace plot training` | KEEP (convenience) |
| `mace_active_learning_md` | `mace md active-learning` | KEEP — uses DEP-1 (#1583) committee support |
| `mace_polar_density_cube` | `mace polar density-cube` | KEEP — electrostatics extra (ELEC-2 (#1592)) |
| `mace_finetuning_select` | `mace train` (with ft config) | MERGE — integrated into the fine-tuning pipeline |
| `mace_convert_device` | *(removed)* | DROP — torch handles device mapping at load; safetensors checkpoints are device-agnostic |
| `mace_e3nn_cueq` | *(removed)* | DROP — backend dispatch is automatic; no per-backend conversion CLI |
| `mace_cueq_to_e3nn` | *(removed)* | DROP — idem |

The three unregistered converter CLIs in `mace/cli/` (`convert_e3nn_oeq`, `convert_oeq_e3nn`, `convert_e3nn_hybrid`) are removed with the same rationale as `mace_e3nn_cueq`, and `mace/cli/visualise_train.py` — an importable module that was never a console script — goes with them. That is **16** modules under `mace/cli/` in total — 12 registered as `console_scripts` plus 4 that never were — and the completeness test cross-references the guide against `git ls-files mace/cli/`, not just against `console_scripts`, so an unregistered module cannot vanish unmentioned.

**C. The DROP list (from the P0-0 inventory) that the guide must enumerate with justifications:**
- **Flags:** `--force_mh_ft_lr` (`arg_parser.py:604`; removed — replay-dependent defaults instead); the **10** legacy "swa" aliases — `--swa` (`:976`), `--start_swa` (`:984`), `--swa_lr` (`:932`), and the seven `--swa_{energy,forces,virials,stress,dipole,polarizability,magforces}_weight` (`:827,802,838,849,860,868,816`) — the old spelling dies with the flag namespace, use the `stage_two` config; `--save_cpu` (`:1064`; safetensors checkpoints are device-agnostic). Note the count: `--swa_magforces_weight` arrived with the magnetic family and makes it ten, not nine. `--config` is MERGED, not dropped: the v1 config system makes config files first-class. So are the three backend-selection flags — `--enable_cueq`, `--enable_oeq` and `--only_cueq` (`:1089`) collapse into one backend-name config field, and `--only_cueq` has no successor of its own because canonical weights remove the build-in-cueq-layout-vs-convert-after-build choice it existed to make.
- **`--model` values:** `BOTNet` (`arg_parser.py:138`) and `ScaleShiftBOTNet` (`:143`) are gone from the v1 model enum. They are DROPs a user can hit today, because argparse accepts both and only `configure_model` refuses them — `RuntimeError("BOTNet is deprecated, use MACE instead")` / `RuntimeError("ScaleShiftBOTNet is deprecated, use MACE instead")` (`mace/tools/model_script_utils.py:374-378`) — so the guide states the replacement the old error already names: **MACE** (`ScaleShiftMACE` for the scale-shifted variant). In v1 the strings fail config validation instead of raising mid-build.
- **Artifacts:** the `_compiled.model` / `_stagetwo_compiled.model` TorchScript side artifacts emitted next to every checkpoint (`run_train.py:1162-1189`) are a recorded DROP (RET-5 (#1601)) — v1 checkpoints are neutral-format only and deployment artifacts come from `mace export`. The bundled `mace/calculators/foundations_models/` artifacts are a DROP too: v1 fetches through the FM-3 (#1588) registry.
- **Flags that are KEEP and must be said to be KEEP**, because their absence from the v1 flag list otherwise reads as a drop: `--lbfgs` (`arg_parser.py:992`) and the whole LBFGS training regime (`mace/tools/train.py:383-384,464`; `run_train.py:933-939` switches the optimizer and `:741,757,777` disable `drop_last` for it) survive as a TRN-1 (#1575) optimizer choice in config; the ASE calculator's graph padding (`MACE_ASE_PAD_NUM_ATOMS` / `MACE_ASE_PAD_NUM_EDGES`) survives as DEP-1 (#1583) config over the DATA-3 (#1571) padding contract; the six MLIAP `MACE_*` runtime env vars survive unchanged (DEP-2 (#1594)).
- **Public Python API:** the old import surface (§2 above) is a deliberate break; the guide records old→new equivalences for the common entry points.
- The full DROP set is sourced from the P0-0 traceability matrix so nothing user-facing disappears silently.

**D. Data contract preserved (do not silently break datasets).** The default property-key names in `mace/tools/default_keys.py` are a **data contract**: every labeled XYZ in the wild uses them. All **13** are KEEP — `REF_energy`, `REF_forces`, `REF_stress`, `REF_virials`, `dipole`, `polarizability`, `head`, `REF_charges`, `total_charge`, `total_spin`, `elec_temp`, and the two the magnetic family added, `REF_magmom` and `REF_magforces`. The `<key>_key` CLI-flag derivation (`DefaultKeys.keydict()`) is part of the contract too. The guide states this so existing datasets load unchanged. If any key is renamed downstream, the guide must publish the explicit old→new mapping.

**E. Where the 0.3.x line lives.** There is no maintenance branch. `main` carries v0.3 and keeps taking fixes through `develop` until v1 is released; after that, the line lives at its final release, pinned from PyPI. The guide states this plainly, and that **no end-of-life date is announced**: users are not being handed a deadline. **Open:** whether any v0.3 fix ships after v1 is released, and if so from where, is not settled and REL-1 should settle it.

**Task:**
1. Write the **migration guide** `v0.x → v1` as versioned in-tree docs, with the loud lede (§A: user checkpoints are not runtime-loadable; only published foundation models convert, one-shot) and sections: model conversion via the FM-1 (#1586) tool with the 25-artifact roster, the CLI migration map (§B), the DROP/MERGE/KEEP list with justifications (§C), the preserved data-key contract (§D), and old→new Python-API equivalences.
2. Author the **AI-assistant migration skill file** (a `SKILL.md`-style, `npx skills add`-installable doc under the agent-skills layout) that encodes the same mapping in an assistant-consumable form, so coding assistants can mechanically rewrite a user's `mace_run_train …` invocation / old import into the v1 equivalent, and know which cases have no automated path (arbitrary checkpoints → retrain).
3. Document the 0.3.x end state in the guide per §E: the line lives on `main` until v1 ships and at its final release thereafter, with no announced end-of-life date and no maintenance branch. Cross-link CLI-1 (#1579)'s `v0.3-args → v1-config` migration helper.
4. Make the guide **findable from the tools users already have**: link it from the repo README, the docs landing page and the PyPI long description, since no installed 0.3.x binary will ever point a user at it.
5. Verify the guide + skill are complete against P0-0: every inventory row with disposition DROP/MERGE/renamed KEEP is addressed, every `setup.cfg` `console_scripts` entry appears in the CLI map, and every module under `mace/cli/` is accounted for even when it was never registered as a script.

**Out of scope:** the actual deletion of legacy modules (RET-1 (#1597)..RET-6 (#1602), deletion-only PRs); the v1.0.0 release pipeline, the 1.0 tag, Zenodo DOI and the `git ls-files mace/ == 0` release gate (REL-2 (#1604)); docs consolidation / retiring the `docs` branch / porting the doc-only commits (REL-3 (#1605)); the `mace model convert --to-basis` and `v0.3-args→v1-config` helper implementations themselves (CLI-1 (#1579)); the converter implementation and foundation-model conversions (FM-1 (#1586)/FM-2 (#1587)).

**Acceptance criteria:**
- [ ] The migration guide states, in its opening, that v0.x user checkpoints are not runtime-loadable and that only the published foundation models convert (one-shot), citing the FM-1 (#1586) tool as the only supported model-conversion path.
- [ ] All 12 `console_scripts` entries appear in the CLI migration map with a v1 equivalent or an explicit "removed" + reason; the 3 unregistered converter CLIs and `visualise_train.py` are covered; a completeness check reads `git ls-files mace/cli/`, not only `setup.cfg`.
- [ ] Every P0-0 inventory row marked DROP or MERGE is enumerated in the guide with its justification; a completeness check (script or test) cross-references the guide against the inventory and passes. For argparse rows the cross-reference is keyed on **dest**, the same key the inventory uses, so an aliased flag is checked once and no dest is checked zero times.
- [ ] The DROP list names **10** swa aliases (including `--swa_magforces_weight`), and the KEEP-but-easily-mistaken-for-dropped items (`--lbfgs` and the LBFGS regime, ASE padding env vars, the six MLIAP env vars) are explicitly listed as KEEP; the two dropped `--model` values (`BOTNet`, `ScaleShiftBOTNet`) are listed with MACE as their replacement.
- [ ] All 13 default property-key names are documented as preserved (unchanged data contract), including `REF_magmom`/`REF_magforces`; any rename ships an explicit old→new mapping.
- [ ] The 25-artifact conversion roster is enumerated; MACE-ANI-CC's drop states that it was bundled in the package and names MACE-OFF as the successor.
- [ ] The AI-assistant migration skill file exists in the installable layout and mechanically maps at least the CLI table and the dropped-flag set; it flags "no automated path" for arbitrary checkpoints.
- [ ] The guide records the 0.3.x end state: `main` until v1 ships, the final release thereafter, no maintenance branch, no end-of-life date announced.
- [ ] The guide is reachable from the repo README, the docs landing page and the PyPI long description — the only signposts that exist, since no shipped 0.3.x binary names v1.

**Verify:**
```bash
# guide/skill completeness against the P0-0 inventory (CPU, no training)
python tests/golden/check_inventory.py                                  # inventory itself still consistent
python -m pytest packages/mace-torch/tests/release/test_migration_guide_complete.py -v   # every DROP/MERGE row, every entry point, every mace/cli module
# deprecation-release signposting
grep -rq '0.3.x' docs/ README.md   # the 0.3.x end state is documented and linked
```
**Review focus:** that the "not runtime-loadable / one-shot convertible only" message is unmistakable and correct, that the CLI map and DROP list are exhaustive against P0-0 (nothing user-facing vanishes silently — the counts 12 scripts / 10 swa aliases / 13 data keys / 25 artifacts are gates, not decoration), and that the preserved data-key contract is accurate.

---
